app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Blukon.java

```diff
@@ -592,15 +603,12 @@ private static void handlegetHistoricDataResponse(byte[] buffer) {
             m_communicationStarted = false;
 
             Log.i(TAG, "Full data that was received is " + HexDump.dumpHexString(m_full_data));
-            LibreBlock.createAndSave("blukon", now, m_full_data, 0);
-            
-            Intent intent = new Intent(Intents.XDRIP_PLUS_LIBRE_DATA);
-            Bundle bundle = new Bundle();
-            bundle.putByteArray(Intents.LIBRE_DATA_BUFFER, m_full_data);
-            bundle.putLong(Intents.LIBRE_DATA_TIMESTAMP, now);
-            intent.putExtras(bundle);
-            intent.addFlags(Intent.FLAG_INCLUDE_STOPPED_PACKAGES);
-            xdrip.getAppContext().sendBroadcast(intent);
+
+            final String tagId = PersistentStore.getString("LibreSN");
+            NFCReaderX.HandleGoodReading(tagId, m_full_data, now);
+
```

Comment:
What happens if tagId is empty here? As can be seen from other parts of blukon.java, we don't always decode the serialnumber:
```
if (JoH.pratelimit(BLUKON_DECODE_SERIAL_TIMER, GET_DECODE_SERIAL_DELAY)) {
                decodeSerialNumber(buffer);
}
```

Reply:
I've run the code a couple of times with an empty tagId, and it doesn't appear to cause any problems (I had initially created a fallback to the string "blukon", and was testing its removal). But perhaps someone with more experience of the LibreAlarm and NFCReaderX code can comment?